extensions/interactions/base.py

```diff
@@ -263,3 +263,18 @@ def get_rule_param_type(self, rule_name, rule_param_name):
                 return param_type
         raise Exception(
             'Rule %s has no param called %s' % (rule_name, rule_param_name))
+
+    def get_hint_proto(self, hints):
```

Comment:
I don't understand why this needs to be here at all? It should be in state_domain.py since it is not differing for each interaction.

Reply:
`state_domain` class Hint already has `to_proto` function. As per earlier @vojtechjelinek suggestions, we should keep the common things from the interactions files to the base file. `get_hint_proto` function will take the hints list and return the list of hint protos.